### PR TITLE
airbuddy: 2.6.3 -> 2.7

### PR DESCRIPTION
**Pull request by stepbrobd**
###### Description of changes

update airbuddy: 2.6.3 -> 2.7

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by StepBroBD** [comment]
On x86_64-darwin and aarch64-darwin:

```shell
$ nix run nixpkgs#nixpkgs-review -- pr 242305
1 package updated:
airbuddy (2.6.3 → 2.7)
1 package built:
airbuddy
```

File: pkgs/os-specific/darwin/airbuddy/default.nix [hunk 1]
```diff
@@ -21,6 +21,24 @@ stdenvNoCC.mkDerivation (finalAttrs: {
 
   nativeBuildInputs = [ undmg ];
 
+  # AirBuddy.dmg is not HFS formatted, default unpackPhase fails
+  # https://discourse.nixos.org/t/help-with-error-only-hfs-file-systems-are-supported-on-ventura
+  unpackCmd = ''
+    if ! [[ "$curSrc" =~ \.dmg$ ]]; then return 1; fi
+    mnt=$(mktemp -d -t ci-XXXXXXXXXX)
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/airbuddy/default.nix:28]:
```suggestion
    mnt=$(mktemp -d)
```

File: pkgs/os-specific/darwin/airbuddy/default.nix [hunk 2]
```diff
@@ -21,6 +21,24 @@ stdenvNoCC.mkDerivation (finalAttrs: {
...
+
+    function finish {
+      /usr/bin/hdiutil detach $mnt -force
+    }
+    trap finish EXIT
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/airbuddy/default.nix:33]: Why do we need the cleanup hook? The sandbox should remove everything anyway

File: pkgs/os-specific/darwin/airbuddy/default.nix [hunk 3]
```diff
@@ -21,6 +21,19 @@ stdenvNoCC.mkDerivation (finalAttrs: {
 
   nativeBuildInputs = [ undmg ];
 
+  # AirBuddy.dmg is not HFS formatted, default unpackPhase fails
+  # https://discourse.nixos.org/t/help-with-error-only-hfs-file-systems-are-supported-on-ventura
+  unpackCmd = ''
+    if ! [[ "$curSrc" =~ \.dmg$ ]]; then return 1; fi
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/airbuddy/default.nix:27]:
```suggestion
```
How should this fail?

**Pull request merged**